### ISSUE TITLE
Question: how to limit posts properly

### ISSUE THREAD

**Issue by mohsenkhanpour** [hexojs/hexo#3415]
Hello everyone, I have been using Hexo recently, it is a very customizable static site generator, kudos to the devs.

I had some posts written in two languages, and I wanted to filter them using the `site.categories` object. However I am seeing some inconsistencies that I would like to get feedback on.

First of all I am able to get all posts in the category `Design` written in the `lang` key in my `languages` (language/en.yml for example)by doing the following:

```
      - var featuredArticles = site.categories.findOne( {name: 'Design'} ).posts;
      
      for post in featuredArticles.toArray()
        if (post.lang == __('lang'))
          //rest of the stuff
```
This returns two posts in the default language as expected (http://localhost:4000/):
![image](https://user-images.githubusercontent.com/28626829/50854074-547e5400-1399-11e9-98a7-51a0a3a6b9f9.png)

and 4 posts in English language (http://localhost:4000/en/)
![image](https://user-images.githubusercontent.com/28626829/50854148-898aa680-1399-11e9-943c-55d08705fdd0.png)

However when adding `.limit(1)` or `.limit(2)` it works on the English index, but not the Arabic one. 

` var featuredArticles = site.categories.findOne( {name: 'Design'} ).posts.limit(1);`

The Arabic one brings an empty page with no posts and no errors in the console. But the limit is applied as expected on the English one (I tried with 1, 2, 3 and it worked flawlessly).

![image](https://user-images.githubusercontent.com/28626829/50854720-18e48980-139b-11e9-9301-0732a9bea046.png)

![image](https://user-images.githubusercontent.com/28626829/50854253-cd7dab80-1399-11e9-8257-30ef9a6e6676.png)

Now I have some questions:
1. What am I doing wrong, and how to fix it?
2. Is there a better way of handling `if (post.lang == __('lang'))`
	I tried 
	`var featuredArticles = site.categories.findOne( {name: 'Design'} ).lang.findOne({name: __('lang')}).posts;`
	and
	`var featuredArticles = site.categories.findOne( {name: 'Design'} ).posts.lang.findOne({name: __('lang')});`
	with no luck.

Thanks for your attention.

**Comment by mohsenkhanpour** [comment]
I found the problem with the Arabic posts not being loaded I removed `if (post.lang == __('lang'))` and it loaded 1 post but it was English. 
So `featuredArticles` had only 1 post and it wasn't English so it didn't show up.
However I still have the second question: How to configure `featuredArticles` in a way that has both `lang` and `limit` at the same time?
Something like:
`var featuredArticles = site.categories.findOne( {name: 'Design'} ).posts.find({lang: "en"}).limit(1);`
